src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CDAConfiguration.java

```diff
@@ -98,4 +99,20 @@ public Optional<URI> getPrivateKeyUri() {
     public Optional<URI> getCertificateUri() {
         return ca.getCertificateUri();
     }
+
+    /**
+     * Verifies if the configuration for the certificateAuthority has changed, given a previous
+     * configuration.
+     *
+     * @param config  CDAConfiguration
+     */
+    public boolean certificateAuthorityChanged(CDAConfiguration config) {
```

Comment:
`equals` ?

Reply:
I like the `certificateAuthorityChanged` equals might be good if we are comparing the whole config.